src/services/hsm.service.js

```diff
@@ -98,40 +308,50 @@ async function isKeyValidXpriv(hsmConnection, hsmKeyName) {
 }
 
 /**
+ * Check that a key exists
+ * @param {hsm.interfaces.Hsm} hsmConnection
+ * @param {string} keyname
+ * @returns {Promise<boolean>}
+ */
+async function hsmKeyExists(hsmConnection, keyname) {
+  try {
+    await hsmConnection.blockchain.getKeyInfo(keyname);
+    return true;
+  } catch (e) {
+    // hsm.constants.ERR_CANNOT_OPEN_OBJ = 5004
+    if (e.errorCode === hsm.constants.ERR_CANNOT_OPEN_OBJ) {
+      await delay(500);
```

Comment:
Can you please add a comment explaining why we need this delay?

Reply:
Done